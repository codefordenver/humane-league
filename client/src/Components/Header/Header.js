import React from 'react';
import { withRouter } from 'react-router';
import thlLogoWhite from '../../THL-Assets/png/THL18-horiz-logo-white.png';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../../Actions/';
import firebase from '../../firebase';

export const Header = ({ User, logout, validateUser, history, location }) => {
  let currentUser = JSON.parse(localStorage.getItem('THL-FAN-USER'));
  
  if (Object.keys(User).length === 0 && currentUser) {
    validateUser(currentUser);
  }

  if (Object.keys(User).length === 0 && !currentUser && location.pathname !== '/' && location.pathname !== '/signin' && location.pathname !== '/forgotpassword') {
    history.push('/');
  }
  
  const userInfo = User.name ? <p className="welcome">Welcome, {User.name.split(' ')[0]}</p> : null;
  const actions = (User.name && !User.admin) ? <NavLink exact to="/home" className="nav-btn"><p>Actions</p></NavLink> : null;
  const profile = (User.name && !User.admin) ? <NavLink exact to="/profile" className="nav-btn"><p>Profile</p></NavLink> : null;
  const adminDash = (User.name && User.admin) ? <NavLink className='nav-btn' exact to='/admin'><p>Dashboard</p></NavLink> : null;
  const adminActions = (User.name && User.admin) ? <NavLink className="nav-btn" exact to="/admin/actions"><p>View Actions</p></NavLink> : null;
  const logoutButton = User.name 
    ? <Link 
      to="/" 
      onClick={() => { 
        firebase.auth().signOut().then(() => {
          logout();
        })
      }}
      className="nav-btn">
      <p className="login-logout-btn">Logout</p></Link> 
    : null;

  return (
    <header>
      <a 
        className='main-site'
        target='_blank' 
        rel='noopener noreferrer' 
        href='https://thehumaneleague.org/'>
        <img 
          className="App__logo" 
          src={thlLogoWhite} />
      </a>
      <nav>
        {userInfo}
        {actions}
        {profile}
        {adminDash}
        {adminActions}
        {logoutButton}
      </nav>
    </header>
  );
}

const mapStateToProps = store => ({
  User: store.User
})

const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(actions.logout()),
  validateUser: user => dispatch(actions.updateUser(user))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header));

