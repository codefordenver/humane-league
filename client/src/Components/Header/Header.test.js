import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header component tests', () => {
  let renderedHeader;
  global.localStorage = {
    'THL-FAN-USER': { id: 1 },
    getItem: () => {
      return JSON.stringify({name: 'Thomas', email: '123@mail.com'});
    }
  };
  
  beforeEach(() => {
    renderedHeader = shallow( <Header validateUser={jest.fn()} User={{}}/> );
    
  })
  it('renders without crashing', () => {
    expect(renderedHeader).toBeDefined();
  });
  it('matches SnapShot', () => {
    expect(renderedHeader).toMatchSnapshot();
  });
});