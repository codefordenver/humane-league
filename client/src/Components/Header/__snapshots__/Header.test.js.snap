// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests matches SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    User={Object {}}
    validateUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="main-site"
          href="https://thehumaneleague.org/"
          rel="noopener noreferrer"
          target="_blank"
>
          <img
                    className="App__logo"
                    src="THL18-horiz-logo-white.png"
          />
</a>,
        <nav>
          
</nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            className="App__logo"
            src="THL18-horiz-logo-white.png"
/>,
          "className": "main-site",
          "href": "https://thehumaneleague.org/",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "App__logo",
            "src": "THL18-horiz-logo-white.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            null,
            null,
            null,
            null,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          null,
          null,
          null,
          null,
          null,
          null,
        ],
        "type": "nav",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="main-site"
            href="https://thehumaneleague.org/"
            rel="noopener noreferrer"
            target="_blank"
>
            <img
                        className="App__logo"
                        src="THL18-horiz-logo-white.png"
            />
</a>,
          <nav>
            
</nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="App__logo"
              src="THL18-horiz-logo-white.png"
/>,
            "className": "main-site",
            "href": "https://thehumaneleague.org/",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "App__logo",
              "src": "THL18-horiz-logo-white.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              null,
              null,
              null,
              null,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            null,
            null,
            null,
            null,
            null,
            null,
          ],
          "type": "nav",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
